### DISCUSSION TITLE
Should we use Travis CI?

### DISCUSSION THREAD

**Discussion by ghost** [NSCC-COGS/Aestheta#50] [General]
Should we use [Travis CI](https://github.com/marketplace/travis-ci/plan/MDIyOk1hcmtldHBsYWNlTGlzdGluZ1BsYW43MA==#pricing-and-setup)? It's apparently free for open source projects. It basically makes sure that what we commit actually runs.

**Comment by kkmcgg** [comment]
I don't know much about Travis CI or how it works - it looks interesting though. I will have a little snoop around. Feel free to investigate how we could put it to use.

I did stumble upon a service for managing deep learning experiments which is similarly free for personal and open source projects which maybe of benefit to us: https://wandb.ai/site

**Comment by ghost** [comment]
Heads up, I just sent a request to install Travis CI / Staging for NSCC-COGS, for the Aestheta repo only. I'm not actually sure where that request for approval goes. @kevinkmcguigan, when you get a moment, could you let me know if you see any authorization requests?

**Comment by kkmcgg** [comment]
Hey @donuty-party - I've look into this a bit more (wandering around on reddit seeing what people have to say about Travis CI) and It seems that we should now be able to replicate most of the functionality of their free tier with the inbuilt [guithub actions](https://github.com/NSCC-COGS/Aestheta/actions/new) section. I have found where I can install Travis CI on our repo however - is there a particular reason why you would like me to follow though with this?

**Reply by ghost** [reply]
Ahh, I started with Travis CI years ago – projects I used, used it – and didn't realize that functionality was built in. That resolves this discussion: the answer is no!

I've set up the Python Package workflow. Next step: fix the build errors!

**Reply by ghost** [reply]
Awesome, it builds now. There was a mistake in requirements.txt and we didn't have any tests for pytest to run. I fixed requirements.txt and commented out the bit that runs pytest.
